docs/02_stratum0_client.md

```diff
@@ -1,136 +1,290 @@
+- `*.pub`: repository’s public key (RSA).
+
+The public key (`repo.organization.tld.crt` + `repo.organization.tld.pub`) is the one that is needed by clients in order to access the repository, so we will need this later on.
+The corresponding private key (`repo.organization.tld.key`) should *not* be shared.
+
+The master key (`repo.organization.tld.masterkey`) is used to sign a whitelist of known publisher certificates.
+This whitelist is, by default, valid for 30 days, so the signing has to be done regularly (for example via a cron job).
+
+Although you can use a different master key per repository, it is recommended to use the same master key
+for all repositories under a single domain, so that clients only need a single public key to access all repositories under this domain.
+For more information, see the CernVM-FS documentation:
+[https://cvmfs.readthedocs.io/en/stable/cpt-repo.html#master-keys](https://cvmfs.readthedocs.io/en/stable/cpt-repo.html#master-keys).
+
+### 2.1.6 Adding files to the repository
 
-For now we are going to use one master key per repository, but in practice it is recommended to use the same master key for all repositories under a single domain, so that clients only need a single public key to access all repositories under this domain. We will explain this in more detail in the advanced section.
-TODO: DO WE WANT TO EXPLAIN THIS?
+A new repository automatically gets a file named `new_repository` in its root (`/cvmfs/repo.organization.tld`). You can add more files by starting and publishing a transaction, which will be explained in more detail in the [3rd hands-on part of this tutorial](04_publishing.md).
 
-### Add some files to the repository
-A new repository automatically gets a file `new_repository` in its root (`/cvmfs/repo.organization.tld`). You can add more files by starting and publishing a transaction, which will be explained in more detail in a later section. For now it is enough to just run the following commands as root:
+For now it is enough to just run the following commands to add a simple `hello.sh` script to your repository.
+
+First, start the transaction via `cvmfs_server transaction <name_of_repo>`:
 
 ```bash
+# Change this to your repository/domain name!
 MY_REPO_NAME=repo.organization.tld
+```
 
+```bash
 sudo cvmfs_server transaction ${MY_REPO_NAME}
+```
 
-# Now make some changes in /cvmfs/${MY_REPO_NAME},
-# e.g. by adding files or directories.
-# If you made $USER the owner of the repository, you can do this without sudo.
+Next, add the file to the repository (in `/cvmfs/${MY_REPO_NAME}`).
+If you made the current user the owner of the repository, you can do this without `sudo` since you have write
+permissions to your repository:
 
+```bash
+echo '#!/bin/bash' > /cvmfs/${MY_REPO_NAME}/hello.sh
+echo 'echo hello' >> /cvmfs/${MY_REPO_NAME}/hello.sh
+chmod a+x /cvmfs/${MY_REPO_NAME}/hello.sh
+```
+
+Complete the transaction by publishing the changes using `cvmfs_server publish <name_of_repo>`:
```

Comment:
I think this last part of the sentence can be removed, otherwise the same thing is mentioned twice.
```suggestion
Complete the transaction by publishing the changes using:
```